components/camera/server_test.go

```diff
@@ -280,6 +280,33 @@ func TestServer(t *testing.T) {
 		test.That(t, err.Error(), test.ShouldContainSubstring, "invalid mime type")
 	})
 
+	t.Run("GetImage with lazy default", func(t *testing.T) {
+		req := pb.GetImageRequest{
+			Name:     testCameraName,
+			MimeType: utils.MimeTypePNG,
+		}
+
+		testCases := []struct {
+			mimeType string
+			img      image.Image
+		}{
+			{mimeType: utils.MimeTypePNG, img: imgPng},
+			{mimeType: utils.MimeTypeJPEG, img: imgJpeg},
+			{mimeType: utils.MimeTypeRawRGBA, img: img},
+		}
+
+		for _, testCase := range testCases {
+			req.MimeType = testCase.mimeType
+			resp, err := cameraServer.GetImage(context.Background(), &req)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, resp.MimeType, test.ShouldEqual, testCase.mimeType)
+			test.That(t, req.MimeType, test.ShouldEqual, utils.WithLazyMIMEType(testCase.mimeType))
+
+			respImage, err := rimage.DecodeImage(context.Background(), resp.Image, testCase.mimeType)
+			test.That(t, respImage, test.ShouldResemble, testCase.img)
```

Comment:
yea, check here that the respImage is of type `LazyEncodedImage` as well

Reply:
In `server.go::GetImage` on line 96 the server encodes images it gets from `ReadImage` as a `[]byte`. So I can't actually check that the `ReadImage` returned a `LazyEncodedImage`. For testing I make sure that the MIME type returned is the same type we requested plus `lazy`. 

Changing `image/woohoo+lazy` to `image/woohoo` on line 269 also ensures that the client does not attempt decode any of the images it receives from the server regardless of if `+lazy` is present. If it did, that test would fail.